ros/src/twist_controller/pid.py

```diff
@@ -20,18 +20,16 @@ def reset(self):
     def step(self, error, sample_time):
         self.last_int_val = self.int_val
 
-        integral = self.int_val + error * sample_time
+        self.int_val = self.int_val + error * sample_time
         derivative = (error - self.last_error) / sample_time
 
         y = self.kp * error + self.ki * self.int_val + self.kd * derivative
         val = max(self.min, min(y, self.max))
 
+        self.last_error = error
         if val > self.max:
             val = self.max
         elif val < self.min:
             val = self.min
```

Comment:
I think l.30 to l.33 are useless.. l.27 already makes sure that val is within [self.min, self.max] range.

So we can safely remove this.

Reply:
good point
just added another commit